inc/integrations/class-coral.php

```diff
@@ -0,0 +1,164 @@
+<?php
+/**
+ * WP Irving integration for Coral.
+ *
+ * @package WP_Irving;
+ */
+
+namespace WP_Irving\Integrations;
+
+use WP_Irving\Singleton;
+
+/**
+ * Class to integrate Coral with Irving.
+ */
+class Coral {
+	use Singleton;
+
+	/**
+	 * The option key for the integration.
+	 *
+	 * @var string
+	 */
+	private $option_key = 'coral';
+
+	/**
+	 * Holds the option values to be set.
+	 *
+	 * @var array
+	 */
+	private $options;
+
+	/**
+	 * Setup the singleton. Validate JWT is installed, and setup hooks.
+	 */
+	public function setup() {
+		// Retrieve any existing integrations options.
+		$this->options = get_option( 'irving_integrations' );
+
+		// Register settings fields for integrations.
+		add_action( 'admin_init', [ $this, 'register_settings_fields' ] );
+
+		$sso_secret = $this->options[ $this->option_key ]['sso_secret'] ?? false;
+
+		if ( ! empty( $sso_secret ) ){
+			// Expose data to the endpoint.
+			add_filter(
+				'wp_irving_data_endpoints',
+				function ( $endpoints ) {
+					$endpoints[] = $this->get_endpoint_settings();
+
+					return $endpoints;
+				}
+			);
+		}
+	}
+
+	/**
+	 * Register settings fields for display.
+	 */
+	public function register_settings_fields() {
+		// Register new fields for the Coral integration.
+		add_settings_field(
+			'wp_irving_coral_sso_secret',
+			esc_html__( 'Coral SSO Secret', 'wp-irving' ),
+			[ $this, 'render_coral_sso_secret_input' ],
+			'wp_irving_integrations',
+			'irving_integrations_settings'
+		);
+	}
+
+	/**
+	 * Render an input for the Coral SSO secret.
+	 */
+	public function render_coral_sso_secret_input() {
+		// Check to see if there is an existing SSO secret in the option.
+		$sso_secret = $this->options[ $this->option_key ]['sso_secret'] ?? '';
+
+		?>
+			<input type="text" name="irving_integrations[<?php echo esc_attr( 'coral_sso_secret' ); ?>]" value="<?php echo esc_attr( $sso_secret ); ?>" />
+		<?php
+	}
+
+	/**
+	 * Get the endpoint settings.
+	 *
+	 * @return array Endpoint settings.
+	 */
+	public function get_endpoint_settings(): array {
+		return [
+			'slug'     => 'validate_sso_user',
+			'callback' => [ $this, 'process_endpoint_request' ],
+		];
+	}
+
+	/**
+	 * Get the data for the Pico user endpoint verification request.
+	 *
+	 * @param \WP_REST_Request $request The request object.
+	 */
+	public function process_endpoint_request( \WP_REST_Request $request ) {
+		// Allow access from the frontend.
+		header( 'Access-Control-Allow-Origin: ' . home_url() );
+
+		$user = $request->get_param( 'user' );
```

Comment:
@joemcgill Great idea! I'm making a ticket for implementing the validation request once we receive the corresponding endpoint/requirements from Pico. I'll add the validation callback as an implementation detail in that ticket so that we know exactly what input we're validating for.